### PR TITLE
Command won`t prompt if mesh is non-existent

### PR DESCRIPTION
**Pull request by mudit-01**
Fixes https://github.com/openservicemesh/osm/issues/4439

Signed-off-by: mudit singh <mudit.singh@india.nec.com>

**Description**:

**Testing done**: Yes

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [X ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

### CODE REVIEW

**Comment by mudit-01** [comment]
Integration Test with Tresor is failing it might be the case that `osm-system namespace` is missing hence throwing an error for it. Is there any workaround to it?

File: cmd/cli/uninstall_mesh.go [hunk 1]
```diff
@@ -94,6 +94,11 @@ func newUninstallMeshCmd(config *action.Configuration, in io.Reader, out io.Writ
 }
 
 func (d *uninstallMeshCmd) run() error {
+	meshInfoList, _ := getMeshInfoList(d.config, d.clientSet)
+	if len(meshInfoList) == 0 {
+		return errors.Errorf("No osm mesh control planes found")
```
Comment (shashankram) [inline cmd/cli/uninstall_mesh.go:99]:
```suggestion
		return errors.Errorf("No osm mesh found")
```

File: cmd/cli/uninstall_mesh_test.go [hunk 2]
```diff
@@ -117,18 +117,8 @@ var _ = Describe("Running the uninstall command", func() {
...
-			It("should prompt for confirmation", func() {
-				Expect(out.String()).To(ContainSubstring("Uninstall OSM [mesh name: testing] in namespace [" + settings.Namespace() + "] ? [y/n]: "))
-			})
-			It("should error", func() {
-				Expect(err).To(MatchError("No OSM control plane with mesh name [testing] found in namespace [" + settings.Namespace() + "]"))
-			})
-			It("should not give a message confirming the successful uninstall", func() {
-				Expect(out.String()).ToNot(ContainSubstring("OSM [mesh name: testing] in namespace [" + settings.Namespace() + "] uninstalled\n"))
-			})
-
```
Comment (shashankram) [inline cmd/cli/uninstall_mesh_test.go:131]: Please do not remove existing tests. You need to extend the existing test for the scenario you are addressing.

**Review by shashankram — CHANGES_REQUESTED** [review]

File: cmd/cli/uninstall_mesh_test.go (hunk 2 above)
Comment (mudit-01) [reply cmd/cli/uninstall_mesh_test.go:131]: hi @shashankram but these tests will become irrelevant, as when we don't have an osm mesh its prompt for the confirmation of its uninstallation so these 3 tests will not be required

Comment (shashankram) [reply cmd/cli/uninstall_mesh_test.go:131]: I see that the test already covers the 2 scenarios - 1. mesh exists, 2. mesh doesn't exist. My bad I missed this.

**Review by shashankram — APPROVED** [review]
Thanks @mudit-01!

**Review by snehachhabria — APPROVED** [review]

**Comment by jaellio** [comment]
> Integration Test with Tresor is failing it might be the case that `osm-system namespace` is missing, hence throwing an error for it. Is there any workaround to it?

Hi @mudit-01! Thanks so much for this change!! I think I found a small fix to work around the failing integration tests.

The test is failing in a clean-up script (clean-kubernetes.sh) we call when running the automated demo (run-osm-demo.sh). The clean-up script calls `bin/osm uninstall mesh -f --mesh-name "$MESH_NAME" --osm-namespace "$K8S_NAMESPACE"` which will now error when there is no mesh installed. A `force` or `f` flag on the `osm uninstall mesh` command hides any errors or output. The slight modification below to your change will return nil instead of an error when no mesh is installed and `force` is true.

**uninstall_mesh.go**
```go
func (d *uninstallMeshCmd) run() error {
    meshInfoList, _ := getMeshInfoList(d.config, d.clientSet)
    if len(meshInfoList) == 0 {
        if d.force {
                return nil
        }
        return errors.Errorf("No osm mesh found")
    }
```

**Comment by mudit-01** [comment]
Thanks, @jaellio for helping in fixing the issue :)

**Comment by mudit-01** [comment]
Is this PR ok to merge @jaellio @shashankram @snehachhabria ?

**Review by jaellio — APPROVED** [review]

**Comment by mudit-01** [comment]
Thanks @jaellio

**Pull request merged**